week7-ts/src/components/Letter.tsx

```diff
@@ -0,0 +1,137 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router';
+import styled from 'styled-components';
+
+import lock from '../assets/lock.png';
+import { LetterInfo, LetterProps } from '../utils/dataType';
+import Modal from './Modal';
+
+export default function Letter(props: LetterProps) {
+  const { letterInfo } = props;
+  const { name, password, content, images, hint } = letterInfo;
+
+  const [isLocked, setIsLocked] = useState<boolean>(true);
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [choiceLetter, setChoiceLetter] = useState<Pick<LetterInfo, 'hint' | 'password'>>();
+  const navigate = useNavigate();
+
+  const handleLockButton = () => {
+    setIsOpen(true);
+    setChoiceLetter({ hint, password });
+  };
+  const handleEditButton = () => {
+    navigate('/edit', { state: letterInfo });
+  };
+
+  return (
+    <>
+      {choiceLetter && (
+        <Modal
+          letterInfo={choiceLetter}
+          isOpen={isOpen}
+          closeModal={() => setIsOpen(false)}
+          checkAnswer={(isCorrect: boolean) => setIsLocked(isCorrect)}
+        />
+      )}
+      <StRoot>
+        {isLocked ? (
+          <LockButton onClick={handleLockButton} />
+        ) : (
+          <StLetter>
+            <StImageList>
+              {images?.map((url) => (
+                <StImage key={url} url={url} />
+              ))}
+            </StImageList>
+            <StName>
+              <b>{name}</b>님이 남긴 편지에요.
```

Comment:
물리적 강조의 `b` 태그보다는 `em` 태그나 `strong` 태그가 더 의미적인 강조를 해준다고 하네!
[참고 아티클](https://m.blog.naver.com/mathesis_time/222010525727)

```suggestion
              <strong>{name}</strong>님이 남긴 편지에요.
```